### PR TITLE
feat: add catch blocks to fetch requests + reporting

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-325] 

### Description

Adds catch blocks to fetch requests + captureMessage for error reporting on Sentry.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- add catch blocks to fetch requests + reporting - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-325]: https://artsyproduct.atlassian.net/browse/PHIRE-325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (add catch blocks to fetch requests + reporting - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 281eeaa43c8bbc0b6e71a58279982bdbc34e5a85
</p>

**Review by MrSltun — APPROVED** [review]
⛑️

File: src/app/utils/PushNotification.ts [hunk 1]
```diff
@@ -79,8 +80,16 @@ export const saveToken = (token: string, ignoreSameTokenCheck = false) => {
...
+
+        let res
+
+        try {
+          res = await fetch(request)
+        } catch (error) {
+          captureMessage(`Push Notification Error while fetching token: ${error}`, "error")
+        }
+
+        const response = await res?.json()
```
Comment (MounirDhahri) [inline src/app/utils/PushNotification.ts:92]: **Q:** any reason why you extracted `res` as a separate variable here? I would be in favor of wrapping both asyc statements with a try catch statement

File: src/app/utils/googleMaps.ts [hunk 2]
```diff
@@ -53,10 +54,14 @@ export const getLocationDetails = async ({
...
-  )
-  const data = await response.json()
+  let response
+
+  try {
+    response = await fetch("https://maps.googleapis.com/maps/api/place/details/json?" + queryString)
+  } catch (error) {
+    captureMessage(`Error fetching getLocationDetails: ${error}`, "error")
+  }
+  const data = await response?.json()
```
Comment (MounirDhahri) [inline src/app/utils/googleMaps.ts:64]: same here

**Review by MounirDhahri — COMMENTED** [review]
thanks for adding this. I see you didn't throw an error on failures and I think that makes sense since we wouldn't want to break the UX here. I would suggest though to show a toast though in case something didn't go as expected. 
For example, you would press on a link that matches `click.artsy.net`, instead of failing silently and not navigating the user, it could be nice to show a hint about it. We can use `showToast` with a red background color

File: src/app/utils/PushNotification.ts (hunk 1 above)
Comment (gkartalis) [reply src/app/utils/PushNotification.ts:92]: ++ makes sense will do that

File: src/app/Components/ShareSheet/ShareSheet.tsx [hunk 3]
```diff
@@ -91,11 +92,17 @@ export const ShareSheet = () => {
...
-    const base64RawData = await resp.base64()
+    try {
+      resp = await RNFetchBlob.config({
+        fileCache: true,
+      }).fetch("GET", currentImageUrl)
+    } catch (error) {
+      captureMessage(`Error fetching RNFetchBlob imgUrl: ${error}`, "error")
+    }
+
+    const base64RawData = await resp?.base64()
```
[the remark this answers is not in the document]
Comment (araujobarret) [reply src/app/Components/ShareSheet/ShareSheet.tsx:105]: Another unrelated point, I see on line `111` we don't need these `RNFetchBlob.fetch` for cases different than `sale`, isn't better to encapsulate it in a function to only return `{ url: base64Data }` once it's the accepted case?

**Review by brainbicycle — APPROVED** [review]
🦺

**Comment by gkartalis** [comment]
I will actually close this one and reopen - refactored variant of this with a wrapper function around fetch that has also captureMessage for sentry, but will take your comments into account and will re-add you as reviewers

**Pull request closed without merging**